float32 theta1
float32 theta2
float32 theta3
float32 vtheta1
float32 vtheta2
float32 vtheta3
string  mode
